l3kernel/testfiles/m3peek001.lvt

```diff
@@ -205,14 +181,21 @@
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
 \char_set_lccode:nn { 32 } { `* }
-\TEST { peek_remove_spaces }
+\TEST { \peek_remove_spaces:n }
   {
     \use:n { \peek_remove_spaces:n { \tl_show:n } ~ \c_space_token } ~ \c_space_tl
     \if_false: { \fi: \peek_remove_spaces:n { \exp_after:wN \tl_show:n \token_to_str:N }  }
     \use:x { \peek_remove_spaces:n { \tex_show:D } \c_catcode_other_space_tl }
     \tex_lowercase:D { \peek_remove_spaces:n { \tex_show:D } ~ }
   }
 
+
+\TEST { \peek_remove_filler:n }
+  {
+    \tl_set:Nx \l_tmpa_tl { \use:n { \scan_stop: \c_space_tl } ~ \scan_stop: } 
+    \use:n { \peek_remove_filler:n { \tex_show:D } ~ \c_space_token } ~ \c_space_tl
+      \l_tmpa_tl _
+  }
```

Comment:
Can you also add a test for `\exp_not:N`, like `\peek_remove_filler:n { \show } \exp_not:N \undefined \relax X`?  I think the code will correctly treat `\exp_not:N \undefined` as `\relax` but I don't remember.

Reply:
I've tested and we _don't_ currently allow `\exp_not:N` here. That happens because we have a two-step process, using `\romannumeral` to deal with expansion and then `\futurelet` to deal with token testing. To allow `\exp_not:N \foo` we'd have to do everything by hand. I think that's not worth it, but I am ready to hear counter-arguments!

Reply:
You are right, I thought that the token remained under the effect of `\noexpand` long enough to be detected (cf how `\exp_after:wN \peek_after:Nw \exp_after:wN ? \exp_not:N \foo` sets `\l_peek_token` to a weird version of `\relax`).  But it seems we cannot do so.